config/v1/types.go

```diff
@@ -80,6 +80,32 @@ type ConfigMapReference struct {
 	Key string `json:"filename,omitempty"`
 }
 
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Build holds cluster-wide information on how to handle builds. The canonical name is `cluster`
+type Build struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec holds user-settable values for the build controller configuration
+	Spec BuildSpec `json:"spec,omitempty"`
+}
+
+type BuildSpec struct {
+	// AdditionalTrustedCA is a reference to a ConfigMap containing additional CAs that
+	// should be trusted during builds for image pushes and pulls.
+	AdditionalTrustedCA ConfigMapReference `json:"additionalTrustedCA,omitempty"`
```

Comment:
please add the config for builddefaulter and buildoverrider while we're introducing this, it'll save me some iteration later.

Reply:
thanks, lgtm pending feedback from @deads2k about the right way to reuse the structs (might need to extract it out to a third package that both build and config can depend on)

Reply:
> @deads2k is the dependency on `build/v1` an issue?

That's fun.  How deep do the references go.  @smarterclayton opinions?

Reply:
@deads2k I think we want this to be a pointer since it's optional (and we need to go fix Image.Spec.AdditionalTrustedCA to also be a pointer).  Agree?  Otherwise the omitempty behavior is not what we want.

Reply:
@deads2k does not agree, so leave it as is (not a pointer).